File: app/src/main/java/io/lerk/lrkFM/version/VersionInfo.java
package io.lerk.lrkFM.version;

import io.lerk.lrkFM.Handler;

/**
 * Version singleton.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
public class VersionInfo {

    private final Version current, latest;

    private static VersionInfo instance;

    private VersionInfo(Version current, Version latest) {
        this.current = current;
        this.latest = latest;
    }

    public Version getCurrent() {
        return current;
    }

    public Version getLatest() {
        return latest;
    }

    public static void parse(Handler<VersionInfo> callback, String current, String latest) {
        if (instance == null) {
            instance = new VersionInfo(Version.fromString(current), Version.fromString(latest));
        }
        callback.handle(instance);
    }

}

File: app/src/main/java/io/lerk/lrkFM/version/VersionInfo.kt
package io.lerk.lrkFM.version



/**
 * Version singleton.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
class VersionInfo private constructor(val current: Version, val latest: Version) {

    companion object {
        private var instance: VersionInfo? = null
        fun parse(callback: (VersionInfo?) -> Unit, current: String, latest: String) {
            if (instance == null) {
                instance = VersionInfo(
                    Version.Companion.fromString(current),
                    Version.Companion.fromString(latest)
                )
            }
            callback(instance)
        }
    }
}
